src/app/clusters/commodity-price-server/commodity-price-server.cpp

```diff
@@ -260,18 +238,23 @@ Instance::GetDetailedPriceRequest(chip::BitMask<CommodityPriceDetailBitmap> deta
 }
 
 CHIP_ERROR
-Instance::GetDetailedForecastRequest(chip::BitMask<CommodityPriceDetailBitmap> details,
-                                     chip::Platform::ScopedMemoryBuffer<Structs::CommodityPriceStruct::Type> & forecastBuffer,
-                                     size_t bufferSize, DataModel::List<const Structs::CommodityPriceStruct::Type> & forecastList,
-                                     bool isEvent, bool isCommand)
+Instance::GetDetailedForecastRequest(BitMask<CommodityPriceDetailBitmap> details,
+                                     Platform::ScopedMemoryBuffer<Structs::CommodityPriceStruct::Type> & forecastBuffer,
+                                     DataModel::List<const Structs::CommodityPriceStruct::Type> & forecastList, bool isCommand)
 {
 
-    size_t count = 0;
+    size_t count      = 0;
+    size_t bufferSize = mPriceForecast.size();
+
+    if (!forecastBuffer.Calloc(bufferSize))
+    {
+        ChipLogError(AppServer, "Memory allocation failed for forecast buffer");
+        return CHIP_ERROR_NO_MEMORY;
+    }
 
     // TODO work out what the max udp packet size is
-    // For events this needs to be about 500
-    // Without events it can be 900
-    size_t kMaxByteCount      = (isEvent) ? 500 : 900;
+    // For commands it can be 900
+    size_t kMaxByteCount      = 900;
```

Comment:
OK.  I am not carefully reviewing this byte-counting logic, because it should all go away anyway.  But as I said before: the actual logic is not correct (e.g. epoch_s needs 6 bytes, not 5).  It's under-counting the actual amount of memory the data structures will use in all sorts of aywa.